### PR TITLE
update Update-AzVm error message

### PR DESCRIPTION
**Pull request by grizzlytheodore**
## Description

updating Update-AzVm error message to be more specific. 
Addressing bug submitted by Damien https://github.com/Azure/azure-powershell/issues/15081

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from the scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** change version of module in pull request

### CODE REVIEW

File: src/Compute/Compute/ChangeLog.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 ## Version 4.30.0
 * Added parameters `PackageFileName`, `ConfigFileName` for `New-AzGalleryApplicationVersion` 
+* Updated `Update-AzVm` to give constructive error messages when empty variables are passed in parameters.
```
Comment (BethanyZhou) [inline src/Compute/Compute/ChangeLog.md:26]: Please insert your change log message between L22-L23

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

**Comment by grizzlytheodore** [comment]
@BethanyZhou change made. ty

File: src/Compute/Compute/ChangeLog.md [hunk 2]
```diff
@@ -20,6 +20,7 @@
 
 -->
 ## Upcoming Release
+* Updated `Update-AzVm` to give constructive error messages when empty variables are passed in parameters.
```
Comment (BethanyZhou) [inline src/Compute/Compute/ChangeLog.md:23]:
```suggestion
* Updated `Update-AzVm` to give constructive error messages when empty variables are passed in parameters. [#15081]
```

**Comment by BethanyZhou** [comment]
Please check following test cases:
- Microsoft.Azure.Commands.Compute.Test.ScenarioTests.PPGTests.TestPPGVM [FAIL] 
- Microsoft.Azure.Commands.Compute.Test.ScenarioTests.DedicatedHostTests.TestDedicatedHostVirtualMachine [

**Comment by isra-fel** [comment]
Adding `ValidateNotNullOrEmpty` means null or empty values are now invalid. Please explain why it is not a breaking change @grizzlytheodore

**Comment by grizzlytheodore** [comment]
@isra-fel  I've removed adding of [validatenotnullorempty] from other parameters except for IdentityId.

This one is not a breaking change because even tho documentation shows as optional, not using the parameter will error out. What we wanted to improve was if identityId is passed in null and the output error was ambiguous. Adding [validatenotnullorempty] gives better error message when the parameter is passed in null value 
image

**Review by isra-fel — APPROVED** [review]
Makes sense. Thank you.

**Pull request merged**